appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/model/security/ScopesImpl.java

```diff
@@ -39,37 +39,44 @@
  */
 package fish.payara.microprofile.openapi.impl.model.security;
 
-import java.util.HashMap;
-import java.util.LinkedHashMap;
 import java.util.Map;
 
 import org.eclipse.microprofile.openapi.models.security.Scopes;
 
-public class ScopesImpl extends LinkedHashMap<String, String> implements Scopes {
+import fish.payara.microprofile.openapi.impl.model.ExtensibleTreeMap;
+
+public class ScopesImpl extends ExtensibleTreeMap<String, Scopes> implements Scopes {
 
     private static final long serialVersionUID = -615440059031779085L;
 
-    protected Map<String, Object> extensions = new HashMap<>();
+    public ScopesImpl() {
+        super();
+    }
+
+    public ScopesImpl(Map<String, ? extends String> scopes) {
+        super(scopes);
+    }
 
     @Override
     public Scopes addScope(String name, String item) {
-        this.put(name, item);
+        this.put(name, item); // this DOES accept null!
```

Comment:
This seems inconsistent but this add method explicitly allows for null while usually the add methods do not allow for null or should ignore it.